app/main.py

```diff
@@ -1,3 +1,13 @@
 def cache(func):
-    # Write your code here
-    pass
+    cache_dict = {}
+
+    def inner(*args):
+        if hash(tuple(args)) in cache_dict:
+            print("Getting from cache")
+            return cache_dict[hash(tuple(args))]
```

Comment:
When you add something to the dict, you don't need to call hash function. Dict does it by itself

Reply:
the goal is to store not the object itself, but its cache, and take the data also by hash. According to the task "You have a big array with data..."

Reply:
Actually, dict doesn't store keys itself. It use a hash of the key to find a proper place for value in the hash table 
You can check [this](https://pastebin.com/82cJiHeL) script. Sizes of dicts are equal in that example